library/core/src/ptr/const_ptr.rs

```diff
@@ -1030,6 +1030,86 @@ impl<T> *const [T] {
     }
 }
 
+#[cfg(not(bootstrap))]
+#[lang = "const_str_ptr"]
+impl *const str {
+    /// Returns the length of a raw string slice.
+    ///
+    /// The returned value is the number of **bytes**, not the number of characters.
+    ///
+    /// This function is safe, even when the raw string slice cannot be cast to a slice
+    /// reference because the pointer is null or unaligned.
```

Comment:
The `because` part here seems makes it seem like those are the only cases in which turning the `*const str` into a `&str` would be UB, but there are others too. For example, if the pointer is perfectly valid but the length is too high. Maybe this statement can be simplified. Maybe something that mentions that the pointer/address is irrelevant.

Reply:
I borrowed this line from `<*const [T]>::len` [documentation](https://doc.rust-lang.org/std/primitive.pointer.html#method.len).

Should I change this in all places to the following:
> This function is safe, even when the raw string slice cannot be safely cast to a slice reference.

Or maybe something else?

Reply:
Ah I see. Then we should change it there as well. That edit sounds good. Thank you.

I suppose another option is to add `, for example` after `slice reference`, or add `(e.g. because ..)` in parentheses. Then there's still some examples, without implying that they're the only cases.